lib/common/mock.c

```diff
@@ -67,29 +67,62 @@ __wrap_calloc(size_t nmemb, size_t size)
 }
 
 
-/* getenv()
+/* getenv(), setenv(), and unsetenv()
  *
- * If pcmk__mock_getenv is set to true, later calls to getenv() must be preceded
+ * If pcmk__mock_env is set to true, later calls to getenv() must be preceded
  * by:
  *
  *     expect_*(__wrap_getenv, name, ...);
  *     will_return(__wrap_getenv, return_value);
  *
+ * Later calls to setenv() must be preceded by:
+ *
+ *     expect_*(__wrap_setenv, name, ...);
+ *     expect_*(__wrap_setenv, value, ...);
+ *     expect_*(__wrap_setenv, overwrite, ...);
+ *
+ * And later calls to unsetenv() must be preceded by:
+ *
+ *     expect_*(__wrap_unsetenv, name, ...);
+ *
  * expect_* functions: https://api.cmocka.org/group__cmocka__param.html
  */
 
-bool pcmk__mock_getenv = false;
+bool pcmk__mock_env = false;
 
 char *
 __wrap_getenv(const char *name)
 {
-    if (!pcmk__mock_getenv) {
+    if (!pcmk__mock_env) {
         return __real_getenv(name);
     }
     check_expected_ptr(name);
     return mock_ptr_type(char *);
 }
 
+// Mocked setenv() and unsetenv() simply check the arguments and return 0
+int
+__wrap_setenv(const char *name, const char *value, int overwrite)
+{
+    if (!pcmk__mock_env) {
+        return __real_setenv(name, value, overwrite);
+    }
+    check_expected_ptr(name);
+    check_expected_ptr(value);
+    check_expected(overwrite);
+    return 0;
+}
+
+int
+__wrap_unsetenv(const char *name)
+{
+    if (!pcmk__mock_env) {
+        return __real_unsetenv(name);
+    }
+    check_expected_ptr(name);
+    return 0;
+}
```

Comment:
Probably so. My thought was to add stuff to the mocked functions on an as-needed basis. The functions that we were going to unit test with mocked `setenv()`/`unsetenv()` didn't check for errors from those system calls.

Now I'm adding those error checks to the validate commit when I push again (see https://github.com/ClusterLabs/pacemaker/pull/2801#discussion_r949286427). So it might be worth incorporating `errno` into the test somehow... even though it doesn't change the return value (`void`) of `pcmk__set_env_option()`. We'll only be printing an error message in `pcmk__set_env_option()`, in case of `(un)setenv()` failure.